presto-orc/src/test/java/com/facebook/presto/orc/IntegerDictionaryProviderTest.java

```diff
@@ -0,0 +1,302 @@
+                        ImmutableList.of(new NodeId(2, 0),
+                                new NodeId(4, 3),
+                                new NodeId(4, 42),
+                                new NodeId(42, 0))
+                },
+        };
+    }
+
+    @DataProvider(name = "dataForDictionaryLoadingTest")
+    public Object[][] dataForDictionaryLoadingTest()
+    {
+        return dataForDictionaryLoadingTests();
+    }
+
+    private Object[][] flattenDataCorruptionCheckTestInput(Object[][] testCases)
+    {
+        int totalSize = 0;
+        for (Object[] testCase : testCases) {
+            totalSize += ((List<NodeId>) testCase[1]).size();
+        }
+        Object[][] result = new Object[totalSize][2];
+        int index = 0;
+        for (int i = 0; i < testCases.length; ++i) {
+            List<NodeId> irrelevantNodes = (List<NodeId>) testCases[i][1];
+            for (NodeId nodeId : irrelevantNodes) {
+                result[index][0] = testCases[i][0];
+                result[index][1] = nodeId;
+                ++index;
+            }
+        }
+        return result;
+    }
+
+    @DataProvider(name = "dataForTestDataCorruptionCheck")
+    public Object[][] dataForTestDataCorruptionCheck()
+    {
+        return flattenDataCorruptionCheckTestInput(dataForDictionaryLoadingTests());
+    }
+
+    @Test(dataProvider = "dataForBufferReuseTest")
+    public void testBufferReuse(long[] buffer, int items, boolean reused)
+            throws IOException
+    {
+        NodeId nodeId = new NodeId(1, 0);
+        long[] data = new long[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16};
+        IntegerDictionaryProvider dictionaryProvider = new IntegerDictionaryProvider(createLongDictionaryStreamSources(ImmutableMap.of(nodeId, data)));
+        StreamId streamId = nodeId.toDictionaryDataStreamId();
+        long[] newDictionary = dictionaryProvider.getDictionary(createFlatStreamDescriptor(streamId), buffer, items);
+        assertEquals(newDictionary == buffer, reused);
+        assertTrue(newDictionary.length >= items);
+    }
+
+    @DataProvider(name = "dataForBufferReuseTest")
+    public Object[][] dataForBufferReuseTest()
+    {
+        return new Object[][] {
+                {new long[0], 4, false},
+                {new long[4], 0, true},
+                {new long[4], 4, true},
+                {new long[4], 8, false},
+                {new long[8], 4, true},
+                {new long[16], 12, true},
+                {new long[16], 16, true},
+        };
+    }
+
+    private static class NodeId
+    {
+        private final int node;
+        private final int sequence;
+
+        public NodeId(int node, int sequence)
+        {
+            this.node = node;
+            this.sequence = sequence;
+        }
+
+        public StreamId toDictionaryDataStreamId()
+        {
+            return new StreamId(node, sequence, DICTIONARY_DATA);
+        }
+    }
+
+    private static class DummyOrcDataSource
```

Comment:
Move static classes to the bottom of the file. As a shorter alternative you can init a TempFile object and then pass it into `OrcTestingUtil.fileOrcDataSource` to get the OrcDataSource.